File: Seccion 22 - Algoritmos Recursivos/Seccion_22_Python/02_ejercicio_02.py

import math

# Solución recursiva
def suma_recursiva(n):
    if n <= 9:
        return n
    else:
        return suma_recursiva(math.floor(n / 10)) + n % 10

# Solución iterativa
def suma_iterativa(n):
    suma = 0
    
    while n > 9:
        suma += n % 10
        n = math.floor(n / 10)
    
    return suma + n

def main():
    
    numero = int(input("Digite un número: "))
    
    print("\nLa suma de los dígitos del número es: ")
    print(f"Algoritmo recursivo: {suma_recursiva(numero)}")
    print(f"Algoritmo iterativo: {suma_iterativa(numero)}")

if __name__ == "__main__":
    main()

File: Seccion 22 - Algoritmos Recursivos/Seccion_22_Dart/02_ejercicio_02.dart

import 'dart:io';

int sumaRecursiva(int n) {
  if (n <= 9) {
    return n;
  } else {
    return (sumaRecursiva((n / 10).floor()) + n % 10);
  }
}

int sumaIterativa(int n) {
  int suma = 0;

  while (n > 9) {
    suma += n % 10;
    n = (n / 10).floor();
  }

  return suma + n;
}

void main() {

  int numero;

  stdout.write("Digite un número: ");
  numero = int.parse(stdin.readLineSync());

  print("\nLa suma de los dígitos del número es: ");
  print("Algoritmo recursivo: ${sumaRecursiva(numero)}");
  print("Algoritmo iterativo: ${sumaIterativa(numero)}");
}